lambdas/data-migration1/tests/test-providers.js

```diff
@@ -0,0 +1,363 @@
+const cryptoRandomString = require('crypto-random-string');
+const omit = require('lodash/omit');
+const path = require('path');
+const test = require('ava');
+
+const Provider = require('@cumulus/api/models/providers');
+const Rule = require('@cumulus/api/models/rules');
+const KMS = require('@cumulus/aws-client/KMS');
+const { dynamodbDocClient } = require('@cumulus/aws-client/services');
+const {
+  createBucket,
+  putFile,
+  recursivelyDeleteS3Bucket,
+} = require('@cumulus/aws-client/S3');
+const { S3KeyPairProvider } = require('@cumulus/common/key-pair-provider');
+const { getKnexClient, localStackConnectionEnv } = require('@cumulus/db');
+
+const {
+  migrateProviderRecord,
+  migrateProviders,
+} = require('../dist/lambda/providers');
+const { RecordAlreadyMigrated } = require('../dist/lambda/errors');
+
+const testDbName = `data_migration_1_${cryptoRandomString({ length: 10 })}`;
+const testDbUser = 'postgres';
+
+const generateFakeProvider = (params) => ({
+  id: cryptoRandomString({ length: 10 }),
+  globalConnectionLimit: 1,
+  protocol: 'http',
+  host: `${cryptoRandomString({ length: 10 })}host`,
+  port: 80,
+  createdAt: Date.now(),
+  updatedAt: Date.now(),
+  username: `${cryptoRandomString({ length: 5 })}user`,
+  password: `${cryptoRandomString({ length: 5 })}pass`,
+  encrypted: false,
+  privateKey: 'key',
+  cmKeyId: 'key-id',
+  certificateUri: 'uri',
+  ...params,
+});
+
+let providersModel;
+let rulesModel;
+
+test.before(async (t) => {
+  t.context.knexAdmin = await getKnexClient({
```

Comment:
This might be helpful: https://github.com/nasa/cumulus/pull/1878/files#diff-69d8e7315b4a49aed756eedad7781a18R7

Reply:
63da80250b13951b2e6723f175eb2b536bcc7bb5